[RDY] Simpler handling for "always-on"/"always-off" options.

fwalch opened the pull request:
Current "always-off" options are `compatible` and `edcompatible`.
Updated the documentation for consistency. `set ttyfast` is no error anymore (but `set nottyfast` is).

justinmk commented:
:+1: Good idea. Is there any way to avoid more globals? Also wondering if we can reduce the code needed in `set_bool_option()`.

Pyrohh commented:
LGTM, my previous solution wasn't exactly elegant :smile:

fwalch committed:
Simpler handling for "always-on"/"always-off" options.

'compatible' and 'edcompatible' are forced to be off.
'ttyfast' is forced to be on.

Pyrohh commented:
I think Travis might have been having issues when you pushed, given that everything failed. I've since rebased and repushed my PRs and they're passing... weird.

elmart commented:
LGTM.

aktau commented:
Great, LGTM!

justinmk commented on runtime/doc/options.txt:
```diff
@@ -7170,6 +7173,9 @@ A jump table for the options with a short description can be found at |Q_op|.
 				     *'ttyfast'* *'tf'* *'nottyfast'* *'notf'*
 'ttyfast' 'tf'		Removed. {Nvim}
 
+	Support for slow terminal connections has been removed.  "set tf" is
```
There should not be an empty line above this, to be consistent with the other entries.

fwalch replied:
I went for consistency with the text at `'compatible'`, but I'll change that, too.

justinmk commented on runtime/doc/options.txt:
```diff
@@ -2417,6 +2417,9 @@ A jump table for the options with a short description can be found at |Q_op|.
 'edcompatible' 'ed'	Removed. {Nvim}
 
+	Support for Ed-compatible mode has been removed.  "set noed" is
+	ignored, and "set ed" is an error.
```
First sentence is redundant with line 2420.

fwalch replied:
I went for consistency with `'compatible'` here as well. Just remove the first sentence entirely?

justinmk replied:
Sure

fwalch replied:
Done.

justinmk commented:
merged, thanks!
